packages/telestion-client-template/template/scripts/register-precommit-hook.js

```diff
@@ -0,0 +1,30 @@
+const util = require('util');
+const child_process = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+const exec = util.promisify(child_process.exec);
+
+async function run() {
+	try {
+		const gitRoot = (await exec('git rev-parse --show-toplevel')).stdout.trim();
+		let hooksFolder = path.join(gitRoot, '.git', 'hooks');
+
+		if (!fs.existsSync(hooksFolder)) {
+			fs.mkdirSync(hooksFolder);
+		}
+
+		const preCommitPath = path.join(hooksFolder, 'pre-commit');
+		if (!fs.existsSync(preCommitPath)) {
+			fs.writeFileSync(
+				preCommitPath,
+				`#!/bin/sh\n./client/node_modules/.bin/pretty-quick --staged --pattern "client/**/*"`
+			);
```

Comment:
Can we convert this file into an ejs-template?
Then we can replace parts of this string, if we're in a telestion-project (with the `client` folder) or in a client with its own repo.

Something like:
```suggestion
			fs.writeFileSync(
				preCommitPath,
				`#!/bin/sh\n<% projectRoot %>/node_modules/.bin/pretty-quick --staged --pattern "<% projectRoot %>/**/*"`
			);
```
Where `<% projectRoot %>` is `./client` for a telestion-project and `.` for an own repo.
(I now, it's really ugly :stuck_out_tongue_winking_eye:)

Reply:
The CLI E2E Test doesn't complain, because `pretty-quick` doesn't throw an error, if the path is not accessible/exists. But it don't actually lint the required files.

Reply:
In that case, I would suggest that we just use `npx` (from the script's folder as CWD), as it (at least, according to its definition) should find the correct path without specifying it further.

The thing is that, since we're in the template, we don't have access to EJS, yet.

Reply:
Resolved with 4b1b0ea and f99431f